public/app/features/explore/state/exploreItem.ts

```diff
@@ -0,0 +1,288 @@
+import { AnyAction } from 'redux';
```

Comment:
Just  a thought (not related to this PR) - if we are doing refactor, maybe we could also improve naming of some of the variables/files. For example `exploreItem` isn't too specific and could make you wonder what exactly it is. Maybe something like `explorePane` would be more semantic and it would use Grafana naming for it.

Reply:
I was actually thinking about using Pane as a name for this instead of Item. Wasn't sure if that would make sense to others but if there are already 2 votes for this I will rename it.